HomeWork2/main.py

```diff
@@ -0,0 +1,135 @@
+
+    def __init__(self, pages):
+        self.url = f'{self.BASE_PART}discover/movie'
+        self.genre_data = self.get_genre_table()
+        self.data = []
+        self.get_data_from_pages(pages)
+        self.genre_table = self.genre_table_with_ids()
+
+    def get_response(self, number_of_pages):
+        params = {
+            'include_adult': False,
+            'include_video': False,
+            'sort_by': 'popularity.desc',
+            'page': number_of_pages
+        }
+        return requests.get(self.url, headers=self.HEADERS, params=params)
+
+    def get_genre_table(self):
+        genre_url = f'{self.BASE_PART}genre/movie/list'
+        return (requests.get(genre_url, headers=self.HEADERS)).json()
+
+    def get_data_from_pages(self, num_of_pages):
+        for i in range(1, num_of_pages + 1):
+            response = self.get_response(i)
+            self.data.extend(response.json()['results'])
+
+    def get_multiple_pages(self):
+        return self.data
+
+    def get_most_popular_title(self):
+        return max((param['popularity'], param['title']) for param in self.data)[1]
+
+    def get_movie_data_with_indexes(self):
+        return self.data[3:19:4]
+
+    def search_films_by_keywords(self, keywords):
+        return [movie['title'] for movie in self.data if
+                all(param.lower() in movie['overview'].lower() for param in keywords)]
+
+    def get_genres_name_by_id(self, ids):
+        return [elem['name'] for elem in self.genre_data['genres'] for genre_id in ids if elem['id'] == genre_id]
+
+    def genre_table_with_ids(self):
+        return {genre['name']: genre['id'] for genre in self.genre_data['genres']}
+
+    def remove_movies_with_unwanted_genres(self, genres):
+        unwanted_genres = [self.genre_table[genre] for genre in genres if genre in self.genre_table]
+        return [elem for elem in self.data if not any(genre_id in elem['genre_ids'] for genre_id in unwanted_genres)]
+
+    def group_movies(self):
+        return [(movie1['title'], movie2['title'])
+                for i, movie1 in enumerate(self.data)
+                for movie2 in self.data[i + 1:]
+                if any(genre in movie2['genre_ids'] for genre in movie1['genre_ids'])]
+
+    def get_genre_ids(self):
+        all_genres_ids = []
+        for genre in self.data:
+            all_genres_ids.extend(genre['genre_ids'])
+        return all_genres_ids
+
+    def get_genre_collection(self):
+        return tuple(self.genre_table_with_ids().keys())
+
+    def get_most_common_genres(self):
+        return dict(Counter(self.get_genres_name_by_id(self.get_genre_ids())))
+
+    def genre_table_replace_first_id_with_22(self):
+        genre_table_modified = self.data
+        genre_table_initial = deepcopy(genre_table_modified)
+        genre_table_modified[0]['genre_ids'][0] = 22
+        print(f'Initial data: {genre_table_initial}')
+        print(f'Modified data: {genre_table_modified}')
```

Comment:
looks weird a bit. why don't you delete **genre_table_initial** variable and just use self.data instead. And create another variable **genre_table_modified** using deep copy.